src/ok.rs

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use super::*;
+use colorful::*;
+use std::fmt;
+
+#[derive(StructOpt, PartialEq)]
+pub enum SevGeneration {
+    #[structopt(about = "Secure Encrypted Virtualization")]
+    Sev,
+
+    #[structopt(about = "SEV + Encrypted State")]
+    Es,
+}
+
+type TestFn = dyn Fn() -> TestResult;
+
+struct Test {
+    name: &'static str,
+    run: Box<TestFn>,
+    sub: Vec<Test>,
+}
+
+struct TestResult {
+    name: &'static str,
+    stat: TestState,
+    mesg: Option<String>,
+}
+
+#[derive(PartialEq)]
+enum TestState {
+    Pass,
+    Skip,
+    Fail,
+}
+
+impl fmt::Display for TestState {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let s = match self {
+            TestState::Pass => format!("{}", " OK ".green()),
```

Comment:
Are the spaces around `OK` necessary? It looks like your formatting function will make sure to center it in a String of at least length 4 and add the spaces automatically.

Reply:
Yea, I noticed this earlier today. I've tried changing the string formatting with different spacing but none seemed to work correctly. Here is the output without the spaces:

```
[ FAIL ] - AMD CPU
[ SKIP ]   - Microcode support
[ SKIP ]   - Secure Memory Encryption (SME)
[ SKIP ]   - Secure Encrypted Virtualization (SEV)
[ SKIP ]     - Physical address bit reduction
[ SKIP ]     - C-bit location
[ SKIP ]     - Number of encrypted guests supported simultaneously
[ SKIP ]     - Minimum ASID value for SEV-enabled, SEV-ES disabled guest
[ SKIP ]     - SEV enabled in KVM
[ SKIP ]     - /dev/sev readable
[ SKIP ]     - /dev/sev writable
[ SKIP ]     - Encrypted State (SEV-ES)
[ SKIP ]   - Page flush MSR
[ OK ] - KVM supported: API version: 12
[ OK ] - Memlock resource limit: Soft: 65536 | Hard: 65536

^^ notice how the two OKs do not align correctly.
```

When adding the spaces, the alignment is as follows:
```
[ FAIL ] - AMD CPU
[ SKIP ]   - Microcode support
[ SKIP ]   - Secure Memory Encryption (SME)
[ SKIP ]   - Secure Encrypted Virtualization (SEV)
[ SKIP ]     - Physical address bit reduction
[ SKIP ]     - C-bit location
[ SKIP ]     - Number of encrypted guests supported simultaneously
[ SKIP ]     - Minimum ASID value for SEV-enabled, SEV-ES disabled guest
[ SKIP ]     - SEV enabled in KVM
[ SKIP ]     - /dev/sev readable
[ SKIP ]     - /dev/sev writable
[ SKIP ]     - Encrypted State (SEV-ES)
[ SKIP ]   - Page flush MSR
[  OK  ] - KVM supported: API version: 12
[  OK  ] - Memlock resource limit: Soft: 65536 | Hard: 65536

^^ alignment is correct
```
Is there any other way to force alignment within the `println!`?

Reply:
I've since changed `OK` to `PASS` to correct the alignment.